users/vieting/experiments/switchboard/ctc/feat/network_helpers/perturbation.py

```diff
@@ -121,17 +123,21 @@ def preemphasis_numpy(waveform, coeff=0.97):
 
     @staticmethod
     def apply_codecs(audio, sample_rate, random_state, codecs):
-        import sox
-
-        tfm = sox.Transformer()
         for codec in codecs:
             prob = codec.pop("prob", 1.0)
             if random_state.random() < prob:
                 if codec.get("encoding") == "ULAW":
-                    tfm.set_output_format(encoding="u-law")
+                    # standard value for mu-law encoding
+                    mu = 255.0
+
+                    # mu-law encoding formula
+                    encoded_audio = np.sign(audio) * np.log1p(mu * np.abs(audio)) / np.log1p(mu)
+                    encoded_audio = ((encoded_audio + 1) / 2 * 255).astype(np.float32)
```

Comment:
Why is this needed?

Reply:
what exactly? This is the mu-law encoding step.

Reply:
The first line (currently 138) corresponds to the encoding equation e.g. on Wikipedia. I'm not sure where you take the second from. That's what my question was about.

Reply:
To move the value range to the same as the  `torchaudio.transforms.MuLawEncoding()` 0-255

Reply:
I wanted to stay consistent with the torchaudio implementation. The signal will be normalized by the wave_norm layer later anyway.

Reply:
This code is supposed to be stand-alone. It should not depend on or make assumptions about what might happen somewhere else. I think if the input is in [-1, 1], it makes more sense to stay in that range for the output as well.

Reply:
Done